overlay.nix

```diff
@@ -0,0 +1,55 @@
+# Overlay function to augment/replace various packages with customized derivations
+self: super:
+let
+  nodeVersions = self.callPackage ./node.nix {};
+  phpVersions = self.callPackage ./php.nix {};
+in
+{
+  # Replace git with a smaller version - this is solely used by npm/composer to fetch
+  # git-based dependencies, so most of the
```

Comment:
Incomplete thought here?

Reply:
No, my thoughts are always

I can't believe that you'd insinuate

I'll push up a